### ISSUE TITLE
Ques about SparseConvTranspose3d and purning

### ISSUE THREAD

**Issue by huixiancheng** [traveller59/spconv#670]
Hi, Yan~！ @traveller59 @FindDefinition 
This issue is about SparseConvTranspose3d.
I wanna use this code to remove redundant voxels. Similar to what you mentioned in #610.
However, I find that in `loss.backward()` stage, the weight of `self.purning` would not update.
Did I make any mistake?
```
class SparseDecoder(spconv.SparseModule):
    expansion = 1

    def __init__(self, inplanes, planes, stride=2, norm_fn=None, indice_key=None):
        super(SparseDecoder, self).__init__()
        assert norm_fn is not None
        bias = norm_fn is not None
        self.deconv = spconv.SparseSequential(
            spconv.SparseConvTranspose3d(inplanes, planes, stride, stride, bias=bias),
            norm_fn(planes),
            nn.ReLU(),
            spconv.SubMConv3d(planes, planes, kernel_size=3, stride=1, padding=1, bias=bias, indice_key=indice_key),
            norm_fn(planes),
            nn.ReLU(),
        )
        self.purning = spconv.SubMConv3d(planes, 1, kernel_size=1, indice_key=indice_key)

    def forward(self, x):
        out = self.deconv(x)

        mask = self.purning(out)
        mask = torch.sigmoid(mask.features) > 0.5
        selected_features = torch.masked_select(out.features, mask).view(-1, out.features.shape[-1])
        selected_indices = torch.masked_select(out.indices, mask).view(-1, out.indices.shape[-1])

        out = replace_feature(out, selected_features)
        out.indices = selected_indices

        # out_tensor = spconv.SparseConvTensor(
        #     features=selected_features,
        #     indices=selected_indices.int(),
        #     spatial_shape=out.spatial_shape,
        #     batch_size=out.batch_size
        # )
        return out

**Comment by huixiancheng** [comment]
I find a similar api in [MinkwowskiEngine](https://github.com/NVIDIA/MinkowskiEngine/blob/02fc608bea4c0549b0a7b00ca1bf15dee4a0b228/examples/reconstruction.py#L456-L468). 
```
        # pruning
        self.pruning = ME.MinkowskiPruning()
```
However, I'm not sure is there a similar implemention in spcov.
